modules/ossm-installing-gateway-using-gateway-injection.adoc

```diff
@@ -0,0 +1,246 @@
+====
+
+. Create a YAML file named `secret-reader.yml` that defines the `ServiceAccount`, `Role`, and `RoleBinding` for the gateway deployment. These settings enable the gateway to read `Secrets` which is required for obtaining TLS credentials.
++
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: secret-reader
+  namespace: <gateway_namespace>
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: secret-reader
+  namespace: <gateway_namespace>
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "watch", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name:  secret-reader
+  namespace: <gateway_namespace>
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: secret-reader
+subjects:
+  - kind: ServiceAccount
+    name:  secret-reader
+----
+
+. Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f secret-reader.yml
+----
+
+. Create a YAML file named `gateway-deployment.yml` that defines the Kubernetes `Deployment` for the gateway. 
++
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: <gateway_name>
+  namespace: <gateway_namespace>
+spec:
+  selector:
+    matchLabels:
+      istio: <gateway_name> 
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway <1>
+      labels:
+        istio: <gateway_name> <2>
+        sidecar.istio.io/inject: "true" <3>
+    spec:
+      containers:
+        - name: istio-proxy
+          image: auto <4>
+          securityContext:
+            capabilities:
+              drop:
+              - ALL
+            allowPrivilegeEscalation: false
+            privileged: false
+            readOnlyRootFilesystem: true
+            runAsNonRoot: true
+          ports:
+          - containerPort: 15090
+            protocol: TCP
+            name: http-envoy-prom
+          resources:
+            limits:
+              cpu: 2000m
+              memory: 1024Mi
+            requests:
+              cpu: 100m
+              memory: 128Mi
+      securityContext:
+        sysctls:
+        - name: net.ipv4.ip_unprivileged_port_start
+          value: "0"
+      serviceAccountName: secret-reader <5>
+----
+<1> Indicates that the {istio} control plane uses the gateway injection template instead of the default sidecar template.
+<2> Ensure that a unique label is set for the gateway deployment. A label is required so that {istio} `Gateway` resources can select gateway workloads. 
+<3> Enables gateway injection by setting the `sidecar.istio.io/inject` parameter to `true`. If the name of the {istio} resource is not `default` you must use the `istio.io/rev: <istio_revision>` label, where the revision represents the active revision of the {istio} resource.
```

Comment:
```suggestion
<3> Enables gateway injection by setting the `sidecar.istio.io/inject` label to `true`. If the name of the {istio} resource is not `default` you must instead use the `istio.io/rev: <istio_revision>` label, where the revision represents the active revision of the {istio} resource.
```

Reply:
I added "instead" because I think some readers could interpret this as meaning "you must _additionally_ use the `istio.io/rev: <istio_revision>` label".

Reply:
ok, thanks for clarifying for me. done.